cluster-autoscaler/simulator/drain.go

```diff
@@ -21,7 +21,7 @@ import (
 	"time"
 
 	apiv1 "k8s.io/api/core/v1"
-	policyv1 "k8s.io/api/policy/v1beta1"
+	policyv1 "k8s.io/api/policy/v1"
```

Comment:
Having the changes between v1 and v1beta1 https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125 seems we need to add check for {} selector in  checkPdbs.

Reply:
Also we need add a test case for empty selector.

Reply:
actually, checkPdbs is already using metav1.LabelSelectorAsSelector(pdb.Spec.Selector), which means a null selector selects nothing and an empty selector `{}` selects everything

that means CA was treating v1beta1 PDBs incorrectly, and is already following v1 semantics, so no changes are needed

Reply:
> Also we need add a test case for empty selector.

I agree with that (and for null selector)